enterprise/frontend/src/embedding-sdk/cli/snippets/nextjs-page-snippet.ts

```diff
@@ -0,0 +1,30 @@
+import path from "path";
+
+import { checkIfUsingAppOrPagesRouter } from "../utils/nextjs-helpers";
+
+// Next.js page for /app/analytics-demo/page.tsx or /pages/analytics-demo.tsx
+export const getNextJsAnalyticsPageSnippet = async (componentPath: string) => {
+  const router = await checkIfUsingAppOrPagesRouter();
```

Comment:
In order to simplify testing for this snippet, please let's pass ALL dynamic things like `router` value or `getImport` function to `getNextJsAnalyticsPageSnippet` as parameters.

During e2e testing it'd be much easier to set proper values for different cases.

Same for all other new snippets

Reply:
@sanex3339 fixed in https://github.com/metabase/metabase/pull/50281/commits/4d800146ef864d2f31f64bed7fca4935825906de, now `resolveImport` is dynamic so it should be easier to work with snippets